tests/test_parse_raw_upload.py

```diff
@@ -0,0 +1,136 @@
+import pytest
+import base64
+import zlib
+import json
+from test_results_parser import parse_raw_upload
+
+class TestParsers:
+    def test_junit(self):
+        with open("tests/junit.xml", "b+r") as f:
+            file_bytes = f.read()
+            raw_upload = {
+                "network": [
+                    "a/b/c.py",
+                ],
+                "test_results_files": [
+                    {
+                        "filename": "junit.xml",
+                        "format": "base64+compressed",
+                        "data": base64.b64encode(zlib.compress(file_bytes)).decode(
+                            "utf-8"
+                        ),
+                    }
+                ]
+            }
+            json_bytes = json.dumps(raw_upload).encode("utf-8")
+            parsing_infos, readable_files_bytes = parse_raw_upload(json_bytes)
+
+
+            readable_files = bytes(readable_files_bytes)
+
+            assert readable_files == f"""# path=junit.xml\n{file_bytes.decode()}\n<<<<<< EOF\n""".encode()
+            
+
+            assert parsing_infos[0]["framework"] == "Pytest"
+            assert parsing_infos[0]["testruns"] == [
```

Comment:
`insta` also exists in Python I believe, if you want to use it for these asserts as well :-)